store/v2/commitment/metadata.go

```diff
@@ -90,12 +104,59 @@ func (m *MetadataStore) flushCommitInfo(version uint64, cInfo *proof.CommitInfo)
 		return err
 	}
 
-	if err := batch.WriteSync(); err != nil {
+	if err := batch.Write(); err != nil {
 		return err
 	}
 	return nil
 }
 
+func (m *MetadataStore) flushRemovedStoreKeys(version uint64, storeKeys []string) (err error) {
+	batch := m.kv.NewBatch()
+	defer func() {
+		err = batch.Close()
+	}()
+
+	for _, storeKey := range storeKeys {
+		key := []byte(fmt.Sprintf("%s%s", encoding.BuildPrefixWithVersion(removedStoreKeyPrefix, version), storeKey))
+		if err := batch.Set(key, []byte{}); err != nil {
```

Comment:
I think if a StoreInfo (below) https://github.com/cosmos/cosmos-sdk/blob/f50b207a031d7b658b18478bd688001c1b177ff3/store/v2/proof/commit_info.go#L24-L28 
was stored here instead of packing all metadata into the key we would gain two key advantages:

1) ability to preload (mount) deleted trees at construction time (i.e. in factory)
2) maintain support for multiple SC types 

In particular this PR makes (2) impossible since `mountTreeFn` is singular.